conans/client/loader.py

```diff
@@ -235,11 +235,7 @@ def layout(_self):
             conanfile.layout = types.MethodType(layout, conanfile)
 
         conanfile.generators = parser.generators
-        try:
-            conanfile.options = Options.loads(parser.options)
-        except Exception:
-            raise ConanException("Error while parsing [options] in conanfile.txt\n"
```

Comment:
This one is necessary, to clarify the error comes from the ``conanfile.txt``, otherwise it is not evident.

The ``ProfileLoader`` will capture the exception and also add information about the offending profile file, this should do the same.